src/backend/distributed/commands/utility_hook.c

```diff
@@ -466,9 +466,30 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 			activeDropSchemaOrDBs++;
 		}
 
+		/*
+		 * Check if we are running ALTER EXTENSION citus UPDATE (TO "<version>") command and
+		 * the available version is different than the current version of Citus. In this case,
+		 * ALTER EXTENSION citus UPDATE command can actually update Citus to a new version.
+		 */
+		bool isAlterExtensionUpdateCitusStmt = isCreateAlterExtensionUpdateCitusStmt &&
+											   IsA(parsetree, AlterExtensionStmt);
+		bool citusCanBeUpdatedToAvailableVersion =
+			!InstalledAndAvailableVersionsCompatible();
```

Comment:
I think we want something like:
```c
bool citusNeedsUpdate = false;

if (IsAlterExtensionCitusUpdate(parstree) &&
    strncmp(InstalledExtensionVersion(), AvailableExtensionVersion(), NAMEDATALEN) != 0)
{
    /* installed version is behind on available version, about to update catalogs */
    citusNeedsUpdate = true;
}


...


if (citusNeedsUpdate && CitusHasBeenLoaded() && IsCoordinator())
{
    /* 
     * Catalogs were updated to the latest version and we're on the coordinator,
     * make sure citus.pg_dist_object contains everything that it would contain
     * if tables/objects had been distributed with the current version of Citus.
     */
    MarkExistingObjectDependenciesDistributedIfSupported();
}


```

Reply:
makes sense